packages/iov-ethereum/src/abi.ts

```diff
@@ -112,6 +117,35 @@ export class Abi {
     return data.slice(32, 32 + length);
   }
 
+  public static decodeSwapProcessState(data: Uint8Array): SwapProcessState {
+    const key = Abi.decodeUint256(data);
+    const map: { readonly [key: string]: SwapProcessState } = {
+      1: SwapProcessState.Open,
+      2: SwapProcessState.Claimed,
+      3: SwapProcessState.Aborted,
+    };
+    const state: SwapProcessState | undefined = map[key];
+
+    if (state === undefined) {
+      throw new Error("Invalid swap process state");
+    }
+    return state;
+  }
+
+  public static decodeEventSignature(data: Uint8Array): SwapContractEvent {
+    const key = Encoding.toHex(data);
+    const map: { readonly [key: string]: SwapContractEvent } = {
+      "22f9086560da4f3a67d5fcc1a440655671d27a7e0884f260be3ce12ead52e156": SwapContractEvent.Opened,
```

Comment:
The key can be replaced by `Encoding.toHex(Abi.calculateMethodHash("Opened(bytes32,address,address,bytes32,uint256,uint256)"))`

Reply:
That's how I calculated it in the first place, but I thought there was no need to recalculate it every time. Do you prefer a comment explaining how it's calculated or just do the calculation each time?

Reply:
What about storing it a module level or class static constants? Then we have readable code and no significant runtime overhead